src/guiguts/mainwindow.py

```diff
@@ -593,6 +582,99 @@ def add_binding(self, key: str, event: str, callback: Callable[[], None]) -> Non
         mouse_bind(self.fields[key], event, lambda *args: callback())
 
 
+class ScrolledReadOnlyText(tk.Text):
+    """Implement a read only mode text editor class with scroll bar.
+
+    Done by replacing the bindings for the insert and delete events. From:
+    http://stackoverflow.com/questions/3842155/is-there-a-way-to-make-the-tkinter-text-widget-read-only
+    """
+
+    def __init__(self, parent, *args, **kwargs):  # type: ignore[no-untyped-def]
+        """Init the class and set the insert and delete event bindings."""
+
+        self.frame = ttk.Frame(parent)
+        self.frame.grid(row=0, column=0, sticky="NSEW")
+        self.frame.columnconfigure(0, weight=1)
+        self.frame.rowconfigure(0, weight=1)
+
+        super().__init__(self.frame, *args, **kwargs)
+        super().grid(column=0, row=0, sticky="NSEW")
+        self.redirector = WidgetRedirector(self)
+        self.insert = self.redirector.register("insert", lambda *args, **kw: "break")
+        self.delete = self.redirector.register("delete", lambda *args, **kw: "break")
+
+        hscroll = ttk.Scrollbar(self.frame, orient=tk.HORIZONTAL, command=self.xview)
+        hscroll.grid(column=0, row=1, sticky="NSEW")
+        self["xscrollcommand"] = hscroll.set
+        vscroll = ttk.Scrollbar(self.frame, orient=tk.VERTICAL, command=self.yview)
+        vscroll.grid(column=1, row=0, sticky="NSEW")
+        self["yscrollcommand"] = vscroll.set
+
+        add_text_context_menu(self, read_only=True)
+
+    def grid(self, *args: Any, **kwargs: Any) -> None:
+        """Override ``grid``, so placing Text actually places surrounding Frame"""
+        return self.frame.grid(*args, **kwargs)
+
+
+class MessageLogDialog(tk.Toplevel):
+    """A Tk simpledialog that displays error/info messages."""
+
+    def __init__(self, parent: tk.Tk, *args: Any, **kwargs: Any) -> None:
+        """Initialize messagelog dialog."""
+        super().__init__(parent, *args, **kwargs)
+        self.title("Message Log")
+        self.columnconfigure(0, weight=1)
+        self.rowconfigure(0, weight=1)
+
+        frame = ttk.Frame(self)
+        frame.columnconfigure(0, weight=1)
+        frame.grid(row=0, column=0, sticky="NSEW")
+        frame.rowconfigure(0, weight=1)
+
+        self.messagelog = ScrolledReadOnlyText(frame, wrap=tk.NONE)
+        self.messagelog.grid(column=0, row=0, sticky="NSEW")
+
+    def append(self, message: str) -> None:
+        """Append a message to the message log dialog."""
+        self.messagelog.insert("end", message)
+
+
+class MessageLog:
```

Comment:
We might want to think about introducing standard python logging in the code and have one of the handlers output errors to the `MessageLogDialog`. That could happen separately after this PR however.

Reply:
I've added an issue #94 to use python logging - looks like just what we need, thanks.